cmd/clusterctl/internal/util/objs.go

```diff
@@ -105,42 +120,87 @@ func FixImages(objs []unstructured.Unstructured, alterImageFunc func(image strin
 	// look for resources of kind Deployment and alter the image
 	for i := range objs {
 		o := &objs[i]
-		if o.GetKind() != deploymentKind {
-			continue
+		err := fixDeploymentImages(o, alterImageFunc)
+		if err != nil {
+			return nil, err
 		}
-
-		// Convert Unstructured into a typed object
-		d := &appsv1.Deployment{}
-		if err := scheme.Scheme.Convert(o, d, nil); err != nil {
+		err = fixDaemonSetImages(o, alterImageFunc)
+		if err != nil {
```

Comment:
```suggestion
		if err := fixDaemonSetImages(o, alterImageFunc); err != nil {
```

Usually we try to avoid shadowing other variables